### DISCUSSION TITLE
need new feature: peer.socket.send

### DISCUSSION THREAD

**Discussion by szcuipeng** [peers/peerjs#1024] [Ideas & Feature requests]
i found peer ( for example: peer = new Peer) owns a websocket connect to peer server, and see the following link:

https://github.com/peers/peerjs-server#combining-with-existing-express-app

the peer server can handle socket request.

right?

then can peerjs provide socket send request/handler response method?

any advices will be appreciated!

**Comment by jonasgloning** [comment]
Hey @szcuipeng. If you listen to the `connection` event on the Express server, you can access the client object including it's websocket.

```ts
const server = PeerServer(opts, (server) => {
  ...
});

server.on("connection", (client) => {
    client.getSocket()?.send("Hello");
});
```

**Reply by szcuipeng** [reply]
on client side, we need peer.on("data"... event to handle "Hello"

**Comment by szcuipeng** [comment]
now in client side:
```ts
let peer = new Peer ( "custom_peer_id" );

peer.on("open",function(peer_id){
    peer._socket._socket.send("json data");
});
```

i'd like to having this:
```ts
peer.getSocket().send("json data");

//and:
peer.on("data",function(data){
    console.log("return from peer server websocket: ",data);
});
```

**Reply by jonasgloning** [reply]
I think we can add this. Someone else already posted an [PR](https://github.com/peers/peerjs/pull/640) for something similar.

**Reply by szcuipeng** [reply]
That's great!
